include/maths/common/CLeastSquaresOnlineRegression.h

```diff
@@ -271,6 +283,46 @@ class CLeastSquaresOnlineRegression : boost::addable<CLeastSquaresOnlineRegressi
         return predict(params, x);
     }
 
+    //! Get the regression model R^2 if it can be computed.
+    //!
+    //! If created with R_2 true then the regression model R^2 can
+    //! be computed from the statistics maintained.
+    //! \param[in] maxCondition The maximum condition number for
+    //! the Gramian this will consider solving. If the condition
+    //! is worse than this it'll fit a lower order polynomial.
+    //! \param[out] result Filled in with the regression model R^2.
+    bool r2(double& result, double maxCondition = MAX_CONDITION) const {
+
+        result = 0.0;
+
+        if (R_2 == false) {
```

Comment:
Could be `if constexpr`

Reply:
See [2ccc477](https://github.com/elastic/ml-cpp/pull/2225/commits/2ccc477a760824de25c3d84cc22649d05c7d4c04).